packages/vite/src/node/utils.ts

```diff
@@ -542,6 +562,18 @@ export function combineSourcemaps(
     return { ...nullSourceMap }
   }
 
+  // hack for parse broken with normalized absolute paths on windows (C:/path/to/something).
+  // escape them to linux like paths
+  sourcemapList.forEach((sourcemap) => {
+    sourcemap.sources = sourcemap.sources.map((source) =>
+      source ? escapeToLinuxLikePath(source) : null
+    )
+    if (sourcemap.sourceRoot) {
+      sourcemap.sourceRoot = escapeToLinuxLikePath(sourcemap.sourceRoot)
+    }
+  })
+  const escapedFilename = escapeToLinuxLikePath(filename)
```

Comment:
Currently, paths used with sourcemap (`sources`/`file`/`sourceRoot` properties) are absolute paths which are obtained from id (`/home/user/path/to/something.js` or on windows `c:/users/user/path/to/something.js`).

There is a problem on windows with this.

```js
const resolve = require('@jridgewell/resolve-uri')

const resolved = resolve('C:/project/foo.sass', 'C:/project/bar.sass')

console.log(resolved) // C:/project/C:/project/foo.sass
```

To solve this I escaped them like `/c/project/foo.sass`.

Reply:
Just trying to understand our options here. Could these paths be made relative to avoid the hack?

Reply:
If there is a file importing a file from a different drive, it would not be able to express in relative paths.
This can happen in the following situation:

- `C:/project/src/foo.css` imports `C:/project/node_modules/pkg/bar.scss`.
- `preserveSymlinks` is false
- `C:/project/node_modules/pkg` is a symbolic link to `D:/somewhere/pkg`

It will [not happen normally for pnpm](https://pnpm.io/faq#does-pnpm-work-across-multiple-drives-or-filesystems) since it does not create cross-drive links.
But if you use `npm i C:/absolute/path/to/pkg` or `yarn add link:C:/absolute/path/to/pkg` or `pnpm link C:/absolute/path/to/pkg`, a symbolic link (on windows accurately a junction) will be created and this can be cross-drive.
Currently I do not have multiple drives in my computer so I did not try it, but it will happen in theory.

Reply:
Also JS sourcemaps uses absolute path.
If CSS sourcemaps will use relative path, it would be better to use relative paths for JS sourcemaps for consistency.

JS sourcemap does need this hack because in JS there is only one input file for one output.